src/main/java/net/snowflake/ingest/streaming/internal/RegisterService.java

```diff
@@ -200,20 +202,25 @@ List<FlushService.BlobData<T>> registerBlobs(Map<String, Timer.Context> latencyT
             this.owningClient.registerBlobs(blobs);
 
             if (registerContext != null) {
-              registerContext.stop();
+              long allBlobRegisterLatencyMs = registerContext.stop();
               blobs.forEach(
                   blob ->
-                      latencyTimerContextMap.computeIfPresent(
+                      flushLatencyTimerContextMap.computeIfPresent(
                           blob.getPath(),
                           (k, v) -> {
-                            v.stop();
+                            long flushLatency = v.stop();
+                            blob.setFlushLatencyMs(flushLatency);
+                            blob.setRegisterLatencyMs(allBlobRegisterLatencyMs);
                             return null;
                           }));
             }
           }
+
+          telemetryService.reportAllBlobLatenciesInSec(blobs, true);
         }
       }
     }
+    // TODO @rcheng question - do we want whatever latencies we can get from error blobs too?
```

Comment:
Unfortunately the errorBlobs don't have the BlobMetadata object, however are the errorBlob latencies (if any) useful for debugging?